js/index.js

```diff
@@ -39,4 +39,81 @@ const siteContent = {
 
 // Example: Update the img src for the logo
 let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+logo.setAttribute('src', siteContent["nav"]["img-src"]);
+
+// Navigation Links
+let navLinks = document.querySelectorAll("nav > a");
+navLinks.forEach((ele, i) => ele.textContent = siteContent["nav"][`nav-item-${i+1}`]);
+
+// CTA Section
+let ctaH1 = document.querySelector(".cta h1");
+ctaH1.textContent = siteContent["cta"]["h1"];
+let ctaBtn = document.querySelector(".cta button");
+ctaBtn.textContent = siteContent["cta"]["button"];
+let ctaImg = document.getElementById("cta-img");
+ctaImg.setAttribute('src', siteContent["cta"]["img-src"]);
+
+// Main Content Section
+let middleImg = document.getElementById("middle-img");
+middleImg.setAttribute('src', siteContent["main-content"]["middle-img-src"]);
+let mainContent = document.querySelectorAll(".main-content .text-content");
+let featuresContent = mainContent[0].children;
+featuresContent[0].textContent = siteContent["main-content"]["features-h4"];
+featuresContent[1].textContent = siteContent["main-content"]["features-content"];
+let aboutContent = mainContent[1].children;
+aboutContent[0].textContent = siteContent["main-content"]["about-h4"];
+aboutContent[1].textContent = siteContent["main-content"]["about-content"];
+let servicesContent = mainContent[2].children;
+servicesContent[0].textContent = siteContent["main-content"]["services-h4"];
+servicesContent[1].textContent = siteContent["main-content"]["services-content"];
+let productContent = mainContent[3].children;
+productContent[0].textContent = siteContent["main-content"]["product-h4"];
+productContent[1].textContent = siteContent["main-content"]["product-content"];
+let visionContent = mainContent[4].children;
+visionContent[0].textContent = siteContent["main-content"]["vision-h4"];
+visionContent[1].textContent = siteContent["main-content"]["vision-content"];
+
```

Comment:
this is good, but it still looks like your can make this shorter and cleaner, a lot of text is being used over and over again, heres one way to fix that:
```
for(let i = 0; i< mainContent.length; i++){
  let section = '';
  switch(i){
    case 0: section ='features'
    case 1: section = 'about'
    case 2: section = 'services'
    case 3: section = 'product'
    case 4: section = 'vision'
  }
  let secArray = mainContent[i].children
  secArr[0].textContent = siteContent["main-content"][`${section}-h4`];
  secArray[1].textContent = siteContent["main-content"][`${section}-content`];
}